src/lib/uTop.ml

```diff
@@ -126,9 +126,21 @@ let get_ocaml_error_message exn =
     Scanf.sscanf
       str
       "Characters %d-%d:\n%[\000-\255]"
-      (fun start stop msg -> ((start, stop), msg))
+      (fun start stop msg -> ((start, stop), msg, None))
   with _ ->
-    ((0, 0), str)
+  try
+    Scanf.sscanf
+      str
+      "Line %d, characters %d-%d:\n%[\000-\255]"
+      (fun line start stop msg -> ((start, stop), msg, Some(line, line)))
+  with _ ->
```

Comment:
Can you catch the specific error here instead?

Reply:
Ack, changing also the untouched for consistency. I used the generic expression because it was used beforehand.